hikari/impl/entity_factory.py

```diff
@@ -470,7 +476,9 @@ def deserialize_audit_log(self, payload: data_binding.JSONObject) -> audit_log_m
                         new_value = value_converter(new_value) if new_value is not None else None
                         old_value = value_converter(old_value) if old_value is not None else None
 
-                    elif not isinstance(key, audit_log_models.AuditLogChangeKey):
+                    elif _LOGGER.isEnabledFor(logging.DEBUG) and not isinstance(
+                        key, audit_log_models.AuditLogChangeKey
+                    ):
```

Comment:
A what about avoiding making multiple unnecessary instance checks every time an entity is deserialized and b isEnabledFor isn't exclusively used for what you described, e.g. :

```json
                # Assume exception for now.
                ex = self.exception()
                self.logger.warning(
                    "encountered unexpected error: %s",
                    ex,
                    exc_info=ex if self.logger.isEnabledFor(logging.DEBUG) else None,
                )
                raise errors.GatewayError("Unexpected websocket exception from gateway") from ex
```

Reply:
> 
> 
> what about avoiding making multiple unnecessary instance checks every time an entity is deserialized? Also isEnabledFor isn't exclusively used for what you described within the library e.g. :
> 
> ```json
>                 # Assume exception for now.
>                 ex = self.exception()
>                 self.logger.warning(
>                     "encountered unexpected error: %s",
>                     ex,
>                     exc_info=ex if self.logger.isEnabledFor(logging.DEBUG) else None,
>                 )
>                 raise errors.GatewayError("Unexpected websocket exception from gateway") from ex
> ```

We use it there to not have logging generate the traceback string and any other info it needs for the exception.

As per the mater in hand, you are right. Even tho this is not called in every entity, it might get called a bit more than we want it to. But I still think that an isinstance check of a class (in this case an enum) will still be faster than checking if the logger is enabled for that level as there are lot of comparisons that happen inside that function

Reply:
> But I still think that an isinstance check of a class (in this case an enum) will still be faster than checking if the logger is enabled for that level as there are lot of comparisons that happen inside that function

That stuff is all cached though, most calls will just be a dict lookup